### PR TITLE
Improved avatar and player name editing

### PR DESCRIPTION
**Pull request by motzel**

### CODE REVIEW

File: src/components/Player/ProfileHeaderInfo.svelte [hunk 1]
```diff
@@ -61,10 +62,15 @@
...
-  function onRedactButtonClick() {
-      if (redactingName && nameInput.value) {
-          account.changeName(nameInput.value);
+  async function onRedactButtonClick() {
+      if (!redactingName) nameInput = name;
+
+      if (redactingName && nameInput !== name) {
+          await account.changeName(nameInput);
+
+          dispatch('player-data-updated', {name: nameInput})
```
Comment (NSGolova) [inline src/components/Player/ProfileHeaderInfo.svelte:71]: A name change can potentially fail here, need to check the `account.error`, or maybe return the error from `account.changeName` function.

Comment (motzel) [reply src/components/Player/ProfileHeaderInfo.svelte:71]: Ok, I found it. I think error handling should be standardized too. Now it just returns a string. IMHO it should be JSON with a fixed format.

Comment (motzel) [reply src/components/Player/ProfileHeaderInfo.svelte:71]: I see it's done 😁
image

Comment (motzel) [reply src/components/Player/ProfileHeaderInfo.svelte:71]: Hmm, not yet xD
image

Comment (motzel) [reply src/components/Player/ProfileHeaderInfo.svelte]: Ok, it's done.

image

**Pull request merged**